app/main.py

```diff
@@ -1,2 +1,124 @@
+from __future__ import annotations
+from typing import List, Tuple, Any, Optional, Hashable
+from dataclasses import dataclass
+
+
+@dataclass
+class Node:
+    __slots__ = ("key", "hash_value", "value")
+
+    key: Hashable
+    hash_value: int
+    value: Any
+
+    def __repr__(self) -> str:
+        return f"<Node: key={self.key} value={self.value}>"
+
+
 class Dictionary:
-    pass
+    __items: List[Optional[Node]]
```

Comment:
why is this class attribute? So all Dictinary objects I create will share the same items..?